### PR TITLE
EPMDJ-5502: update header on tests to run page

### PR DESCRIPTION
**Pull request by IliaSolovev**
Issue: EPMDJ-5502

### CODE REVIEW

File: src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/saved-time-section/saved-time-section.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+const savedTimeSection = BEM(styles);
+
+export const SavedTimeSection = savedTimeSection(
+  ({
+    className, label, percentage, message, children, previousBuildAutoTestsCount,
+  }: Props) => (
+    <div className={className}>
+      <Content data-test={`information-section:${label}`}>
+        <Tooltip message={message && <Message>{message}</Message>} position="top-center">
+          <Title>{ label }</Title>
```
Comment (RomanAchilov) [inline src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/saved-time-section/saved-time-section.tsx:25]:
```suggestion
          <Title>{label}</Title>
```

File: src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.module.scss [hunk 2]
```diff
@@ -36,4 +41,8 @@
     font-size: 12px;
     font-family: $regular-font;
   }
+
+  &__get-suggested-tests-button {
```
Comment (RomanAchilov) [inline src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.module.scss:45]: is this necessary?

File: src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx [hunk 3]
```diff
@@ -5,23 +5,45 @@ import { useParams } from 'react-router-dom';
...
 import { useAgent, useBuildVersion } from 'hooks';
 import { ParentBuild } from 'types/parent-build';
+import { TestCoverageInfo } from 'types/test-coverage-info';
+import { convertToPercentage, getDuration, percentFormatter } from 'utils';
 import { GetSuggestedTestsModal } from './get-suggested-tests-modal';
+import { SavedTimeSection } from './saved-time-section';
 
 import styles from './tests-to-run-header.module.scss';
 
+interface TestsDuration {
```
Comment (RomanAchilov) [inline src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx:15]: move to types folder

File: src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx [hunk 4]
```diff
@@ -5,23 +5,45 @@ import { useParams } from 'react-router-dom';
...
+  count?: number;
+}
+
 interface Props {
   className?: string;
   testsToRunCount: number;
   agentType: string;
+  previousBuildAutoTestsCount?: number;
+  previousBuildTestsDuration: number;
+  autoTests: TestCoverageInfo[];
```
Comment (RomanAchilov) [inline src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx:28]: it looks like this can be generalized

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

File: src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx [hunk 5]
```diff
@@ -60,5 +121,31 @@ export const TestsToRunHeader = testsToRunHeader(({ className, testsToRunCount,
 const Title = testsToRunHeader.title('div');
 const Count = testsToRunHeader.count('div');
 const SubTitle = testsToRunHeader.subTitle('div');
+const Actions = testsToRunHeader.actions('div');
 const CurrentBuildVersion = testsToRunHeader.currentBuildVersion('span');
 const ComparedBuildVersion = testsToRunHeader.comparedBuildVersion('span');
+const GetSuggestedTestsButton = testsToRunHeader.getSuggestedTestsButton(Button);
+
+function getTooltipMessage({
```
Comment (pavyarov) [inline src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx:129]: Looks rly strange, you changing render logic depending at strings.
@RomanAchilov, @RomanDavlyatshin  What do u think?

File: src/utils/index.ts [hunk 6]
```diff
@@ -14,3 +14,4 @@ export { transformObjectsArrayToObject } from './transform-objects-array-to-obje
 export { formatMsToDate } from './format-ms-to-date';
 export { inputLengthRestriction } from './input-length-restriction';
 export { convertToPercentage } from './convert-to-percentage';
+export { getDuration } from './getDuration';
```
Comment (pavyarov) [inline src/utils/index.ts:17]: File name should be at kebab-case

File: src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx [hunk 7]
```diff
@@ -40,16 +62,55 @@ export const TestsToRunHeader = testsToRunHeader(({ className, testsToRunCount,
...
+              previousBuildAutoTestsCount={previousBuildAutoTestsCount}
+              label="total time saved"
+              percentage={autoTests.length && !autoTestsToRun
+                ? percentFormatter(convertToPercentage(previousBuildTestsDuration - currentDuration, previousBuildTestsDuration))
+                : undefined}
+              message={getTooltipMessage({
+                autoTestsToRun, previousBuildAutoTestsCount, label: 'total time saved', autoTestsCount: autoTests.length,
+              })}
+            >
+              { autoTests.length && !autoTestsToRun
```
Comment (pavyarov) [inline src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx:109]: 3 props depending at `autoTests.length` maybe put all this logic to component?

File: src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx (hunk 5 above)
Comment (IliaSolovev) [reply src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx]: i removed this function

File: src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/saved-time-section/saved-time-section.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+export const SavedTimeSection = savedTimeSection(
+  ({
+    className, label, percentage, message, children, previousBuildAutoTestsCount,
+  }: Props) => (
+    <div className={className}>
+      <Content data-test={`information-section:${label}`}>
+        <Tooltip
+          message={message && <Message>{message}</Message>}
+          position={label === 'total time saved' ? 'top-left' : 'top-center'}
```
Comment (RomanAchilov) [inline src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/saved-time-section/saved-time-section.tsx:26]: delete

File: src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx [hunk 9]
```diff
@@ -5,23 +5,46 @@ import { useParams } from 'react-router-dom';
...
+  const {
+    currentDuration = 0, durationByParent = 0,
+  } = useBuildVersion<TestsDuration>('/build/summary/tests-to-run') || {};
   const { buildVersion = '', agentId = '' } = useParams<{ buildVersion: string; agentId: string; }>();
   const { buildVersion: activeBuildVersion = '' } = useAgent(agentId) || {};
+  const { version: previousBuildVersion = '' } = useBuildVersion<ParentBuild>('/data/parent') || {};
+  const { testDuration: previousBuildTestsDuration = 0, tests: previousBuildTests = [] } = useBuildVersion<BuildSummary>(
+    '/build/summary', undefined, undefined, undefined, previousBuildVersion,
+  ) || {};
+  const { AUTO } = previousBuildTests
```
Comment (RomanAchilov) [inline src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx:39]:
```suggestion
  const { AUTO: { total }} = useBuildVersion<TestsToRunSummary>('/build/summary/tests-to-run') || {};
```

File: src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx [hunk 10]
```diff
@@ -40,16 +63,50 @@ export const TestsToRunHeader = testsToRunHeader(({ className, testsToRunCount,
...
+            >
+              {totalDuration.hours}:{totalDuration.minutes}:{totalDuration.seconds}
+            </SavedTimeSection>
+            <SavedTimeSection
+              previousBuildAutoTestsCount={previousBuildAutoTestsCount}
+              label="estimated time saved"
+              percentage={autoTests.length
+                ? percentFormatter(convertToPercentage(previousBuildTestsDuration - durationByParent, previousBuildTestsDuration))
+                : undefined}
+              message={getEstimatedTimeSavedTooltipMessage(autoTests.length, previousBuildAutoTestsCount)}
```
Comment (RomanAchilov) [inline src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx:91]:
```suggestion
              message={prev ? fun() : null}
```

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

File: src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx [hunk 11]
```diff
@@ -42,16 +57,50 @@ export const TestsToRunHeader = testsToRunHeader(({ className, testsToRunCount,
...
+                ? `${estimatedTimeSaved.hours}:${estimatedTimeSaved.minutes}:${estimatedTimeSaved.seconds}`
+                : '––:––:––'}
+            </SavedTimeSection>
+            <SavedTimeSection
+              previousBuildAutoTestsCount={previousBuildAutoTestsCount}
+              label="total time saved"
+              percentage={autoTestsToRun?.total && autoTestsToRun?.total === autoTestsToRun?.completed
+                ? percentFormatter(convertToPercentage(previousBuildTestsDuration - currentDuration, previousBuildTestsDuration))
+                : undefined}
+              message={getTotalSavedTimeTooltipMessage(Number(autoTestsToRun?.total) - Number(autoTestsToRun?.completed))}
```
Comment (RomanAchilov) [inline src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx:97]: why need Number() ?

Comment (RomanAchilov) [reply src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx:97]: set the default values

File: src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx [hunk 12]
```diff
@@ -62,5 +111,23 @@ export const TestsToRunHeader = testsToRunHeader(({ className, testsToRunCount,
...
+    : <span>No data about Auto tests duration in the parent build</span>;
+}
+
+function getEstimatedTimeSavedTooltipMessage(autoTestsCount: number) {
+  return autoTestsCount
+    ? <span>The sum of the suggested Auto tests<br /> duration in the parent build</span>
+    : <span>No suggested Auto tests to run in current build</span>;
+}
+
+function getTotalSavedTimeTooltipMessage(autoTestsToRunCount: number) {
```
Comment (RomanAchilov) [inline src/modules/tests-to-run/tests-to-run-list/tests-to-run-header/tests-to-run-header.tsx:130]: this is strange

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

**Review by RomanAchilov — APPROVED** [review]

**Pull request merged**